src/OSPSuite.R/Services/SimulationAnalyzer.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Serialization.SimModel.Services;
+
+namespace OSPSuite.R.Services
+{
+   public interface ISimulationAnalyzer
+   {
+      IReadOnlyList<string> AllPathOfParametersUsedInSimulation(IModelCoreSimulation modelCoreSimulation);
+   }
+
+   public class SimulationAnalyzer : SimModelManagerBase, ISimulationAnalyzer
+   {
+      public SimulationAnalyzer(ISimModelExporter simModelExporter, ISimModelSimulationFactory simModelSimulationFactory) : base(simModelExporter, simModelSimulationFactory)
+      {
+      }
+
+      public IReadOnlyList<string> AllPathOfParametersUsedInSimulation(IModelCoreSimulation modelCoreSimulation)
+      {
+         var simulationExport = CreateSimulationExport(modelCoreSimulation, SimModelExportMode.Optimized);
```

Comment:
@Yuri05 I have no idea what to do. This seems to always return everything (e.g. all VF parameters even tho they are not used)

Reply:
Ok, I know why. Per default identification of used parameters is disabled in SimModel (because this information was not used by PK-Sim/MoBi but only by the Toolboxes).
In order to activate it, you must set [SimModelSimulation.Options.IdentifyUsedParameters](https://github.com/Open-Systems-Pharmacology/OSPSuite.SimModel/blob/2d478758d11d8f9f8df234701a5624108e434784/src/OSPSuite.SimModel/SimulationOptions.cs#L134) to true before loading SimModel-Simulation from XML.
E.g. you could define `identifyUsedParameters `as an optional parameter for the `CreateSimulation `function (default: false) and [set it before loading:](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/06ec0ffe87fbd46cdee8a6c5c3580fad905f6dd1/src/OSPSuite.Core/Domain/Services/SimModelManagerBase.cs#L47)
```
    var simulation = _simModelSimulationFactory.Create();
    simulation.Options.IdentifyUsedParameters = identifyUsedParameters;
    simulation.LoadFromXMLString(simulationExport);
```

Reply:
Guess one day we should move this functionality from SimModel to OSPSuiteCore (especially if we would like to mark/hide unused parameters in PK-Sim/MoBi)

Reply:
ok/ What's the difference betweeen variableParmaeters and ParameterProperties?

Yes. This would be useful to set this property after the first run for example. One could even hide those parameters in PKSim or mark them with a grayed out color (e..g unused)

Reply:
ParameterProperties are all parameters, VariableParameters are those which can be varied between simulation runs (are set between Load and Finalize)